command/agent/consul/check_watcher.go

```diff
@@ -22,11 +22,21 @@ type ChecksAPI interface {
 	Checks() (map[string]*api.AgentCheck, error)
 }
 
-// TaskRestarter allows the checkWatcher to restart tasks.
-type TaskRestarter interface {
+// WorkloadRestarter allows the checkWatcher to restart tasks or entire task groups.
+type WorkloadRestarter interface {
 	Restart(ctx context.Context, event *structs.TaskEvent, failure bool) error
 }
 
+func NoopRestarter() WorkloadRestarter {
```

Comment:
Is this necessary anymore outside of tests? Maybe move it to a testing.go file if it's needed for multiple package's tests.